core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -60,13 +62,17 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
   override def tearDown() {
     consumer1.close()
     consumer2.close()
+    producers.foreach(_.close())
```

Comment:
@ijuma Yes, this is a fallback to ensure that all producers do get closed. It felt better to close in the teardown rather than add `close()` to the individual tests in case new tests were added. Since `close` is idempotent, I didn't think it was necessary to remove the existing close statements, but I am happy to remove them if it is likely to look confusing. I was aiming for the smallest fix :-)